### PR TITLE
[moveonly] Extract HelpRequested to dry up the help options testing

### PR DESCRIPTION
**Pull request by Empact**
This ensures consistency across interfaces and makes the version handling more clear.

### CODE REVIEW

**Comment by promag** [comment]
utACK 9a7be59.

Just want to point that `HelpRequested` is only called 5 times, IMO it is worth doing this.

Just want to throw two ideas:
 - how about `gArgs.IsHelpArgSet()` or alike?
 - `HelpRequested()` without arguments since `gArgs`  is global?

**Comment by Empact** [comment]
I could go either way on both points. My main inclination is towards consolidation.

**Comment by jnewbery** [comment]
utACK 9a7be5971a0794bc7e6338443add75336f2efd91.

> I could go either way on both points.

I feel the same way.

File: src/util.cpp [hunk 1]
```diff
@@ -584,7 +584,9 @@ void ArgsManager::ForceSetArg(const std::string& strArg, const std::string& strV
     mapMultiArgs[strArg] = {strValue};
 }
 
-
+bool HelpRequested(const ArgsManager& args) {
+  return args.IsArgSet("-?") || args.IsArgSet("-h") || args.IsArgSet("-help");
```
Comment (ryanofsky) [inline src/util.cpp]: Indentation is off (should use 4 spaces)

**Review by ryanofsky — COMMENTED** [review]
utACK 9a7be5971a0794bc7e6338443add75336f2efd91

File: src/util.cpp (hunk 1 above)
Comment (MarcoFalke) [reply src/util.cpp]: Note: Don't forget to install `clang-format` and run the https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools#clang-format-diffpy script to preempt whitespace nitpicking.

Comment (Empact) [reply src/util.cpp]: Fixed, and thanks - I need to set up a hook for that. :P

**Review by ryanofsky — COMMENTED** [review]
utACK b386970d0750e368d6adc5d0c010ead329b5c7c1. Just whitespace changes since last review.

**Comment by MarcoFalke** [comment]
utACK b386970d0750e368d6adc5d0c010ead329b5c7c1

**Pull request merged**